src/extension_utils/utils.ts

```diff
@@ -372,9 +418,194 @@ export const installPythonDependencies = async (
                 }
             });
 
+        return pythonExecutable;
+    }
+
+    return pythonPath;
+};
+
+export const checkForDependencies = async (
+    context: vscode.ExtensionContext,
+    pythonPath: string
+) => {
+    const dependencyCheckerPath: string = getPathToScript(
+        context,
+        CONSTANTS.FILESYSTEM.OUTPUT_DIRECTORY,
+        HELPER_FILES.CHECK_PYTHON_DEPENDENCIES
+    );
+    try {
+        const { stdout } = await executePythonCommand(
+            pythonPath,
+            `"${dependencyCheckerPath}"`
+        );
+        console.info(stdout);
+        return true;
+    } catch (err) {
+        return false;
+    }
+};
+
+export const installDependencies = async (
+    context: vscode.ExtensionContext,
+    pythonPath: string,
+    failSilently: boolean = true
+) => {
+    const requirementsPath: string = getPathToScript(
+        context,
+        CONSTANTS.FILESYSTEM.OUTPUT_DIRECTORY,
+        "requirements.txt"
+    );
+
+    if (!checkForPip(pythonPath)) {
+        return false;
+    }
+
+    try {
+        const { stdout } = await executePythonCommand(
+            pythonPath,
+            `-m pip install -r "${requirementsPath}"`
+        );
+
+        console.info(stdout);
+        vscode.window.showInformationMessage(CONSTANTS.INFO.SUCCESSFUL_INSTALL);
+        return true;
+    } catch (err) {
+        if (!failSilently) {
+            vscode.window
+                .showErrorMessage(
+                    CONSTANTS.ERROR.DEPENDENCY_DOWNLOAD_ERROR,
+                    DialogResponses.READ_INSTALL_MD
+                )
+                .then((selection: vscode.MessageItem | undefined) => {
+                    if (selection === DialogResponses.READ_INSTALL_MD) {
+                        open(CONSTANTS.LINKS.INSTALL);
+                    }
+                });
+        }
+
         console.error(err);
         this.logToOutputChannel(errorChannel, err.toString(), true /* show */);
-        installed = false;
+        return false;
     }
-    return installed;
+};
+
+export const setupEnv = async (
```

Comment:
nit: I would make this method more descriptive somehow.
e.g. you can break it down as `GetPythonExecutableName(...)` and then `SetupEnv(...)`